### DISCUSSION TITLE
What regressors/external feature support would you like supported?

### DISCUSSION THREAD

**Discussion by winedarksea** [winedarksea/AutoTS#205] [Ideas]
Here is the question, if you could have anything, what sort of external information input would you like to have?

_Discussion:_
Right now, AutoTS supports two types of external features:

- future_regressors
- covariates passed as multivariate time series (perhaps with targeted weighting), only used by some multivariate models

The most common type of external feature is a "covariate" or a series which we don't know the future about, but still want to use. Behind the scenes, many models may shift/lag this forward, which is enabled externally here by `create_regressor`, although that functionality confuses some people.

I think some people might like the numerous data input options like some of the neural network forecasting packages have, believing each of these gets treated in unique, special, and highly useful ways behind the scenes (even though that really isn't the case). 

And then there are the users who think they need to make up an X feature set like this were FLAML or some other ml optimizer. Not sure how to unconfuse them.

And another challenge is all the underlying models have very different limitations on what they can accept as inputs. Making something special for the input that only one model can handle
I expect to leave this question open for quite a while, so don't be shy to answer anytime.

**Comment by LukeAFullard** [comment]
I'd be interested to see if you could include [conformal predicton](https://github.com/valeman/awesome-conformal-prediction#papers-time-series) to generate prediction intervals.

**Reply by winedarksea** [reply]
This isn't really related to the discussion question asked...
That said, some of the probabilistic predictions here resemble conformal predictions. But any statistical theory is useless compared to actual cross validation. If you want solid probabilistic bounds, you use SPL (or perhaps containment/coverage) in the metric_weighting and let it choose models that get it right.

**Comment by rcyost** [comment]
I think the naming convention could be much clearer, you have currently use future regressors, regressors, covariates, features, external features, ... . Stick to a naming convention, provide clear definitions, and only use those.
